tests/Data/Test_Contract.cpp

```diff
@@ -427,4 +429,128 @@ BOOST_AUTO_TEST_CASE(testPingPong) {
   /* ------------------------------------------------------------------- */
 }
 
+BOOST_AUTO_TEST_CASE(testFungibleToken) {
+  // 1. Bootstrap our test case. Use test case 0 as the basis.
+  KeyPair owner(priv1, {priv1}), transferee(priv2, {priv2});
+  Address ownerAddr, transfereeAddr, contrAddr;
+  uint64_t nonce = 0;
+
+  INIT_STDOUT_LOGGER();
+
+  LOG_MARKER();
+
+  if (SCILLA_ROOT.empty()) {
+    LOG_GENERAL(WARNING, "SCILLA_ROOT not set to run Test_Contract");
+    return;
+  }
+
+  AccountStore::GetInstance().Init();
+
+  const uint128_t bal{std::numeric_limits<uint128_t>::max()};
+
+  ownerAddr = Account::GetAddressFromPublicKey(owner.second);
+  transfereeAddr = Account::GetAddressFromPublicKey(transferee.second);
+
+  AccountStore::GetInstance().AddAccount(ownerAddr, {bal, nonce});
+  AccountStore::GetInstance().AddAccount(transfereeAddr, {bal, nonce});
+
+  const unsigned int numHodlers[] = {100000, 200000, 300000, 400000, 500000};
+
+  for (auto hodlers : numHodlers) {
+    contrAddr = Account::GetAddressForContract(ownerAddr, nonce);
+    LOG_GENERAL(INFO, "FungibleToken Address: " << contrAddr);
+
+    // Deploy the contract using data from the 1st Scilla test.
+    ScillaTestUtil::ScillaTest t1;
+    if (!ScillaTestUtil::GetScillaTest(t1, "fungible-token", 2)) {
+      LOG_GENERAL(WARNING, "Unable to fetch test fungible-token_1.");
+      return;
+    }
+
+    // Replace owner address in init.json.
+    for (auto& it : t1.init) {
+      if (it["vname"] == "owner") {
+        it["value"] = "0x" + ownerAddr.hex();
+      }
+    }
+    // and remove _creation_block (automatic insertion later).
+    ScillaTestUtil::RemoveCreationBlockFromInit(t1.init);
+
+    uint64_t bnum = ScillaTestUtil::GetBlockNumberFromJson(t1.blockchain);
+
+    // Transaction to deploy contract.
+    std::string initStr = JSONUtils::convertJsontoStr(t1.init);
+    bytes data(initStr.begin(), initStr.end());
+    Transaction tx0(1, nonce, NullAddress, owner, 0, PRECISION_MIN_VALUE,
+                    500000, t1.code, data);
+    TransactionReceipt tr0;
+    AccountStore::GetInstance().UpdateAccounts(bnum, 1, true, tx0, tr0);
+    Account* account = AccountStore::GetInstance().GetAccount(contrAddr);
+    // We should now have a new account.
+    BOOST_CHECK_MESSAGE(account != nullptr,
+                        "Error with creation of contract account");
+    nonce++;
+
+    // 2. Pre-generate and save a large map and save it to LDB
+    std::string initOwnerBalance;
+    for (unsigned int i = 0; i < hodlers; i++) {
+      unsigned char* hodler = new unsigned char[20];
```

Comment:
I don't see `hodler` being deleted anywhere. Maybe you can just have a stack vector (`hodlerAddress` that you already have below) and pass it's `data()` to `RAND_Bytes`.